src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ReferralHelper.java

```diff
@@ -0,0 +1,352 @@
+
+        if (earliestFutureHearingDate == null) {
+            return "None";
+        }
+
+        Date hearingStartDate;
+        try {
+            hearingStartDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS").parse(earliestFutureHearingDate);
+        } catch (ParseException e) {
+            log.info("Failed to parse hearing date when creating new referral");
+            return "None";
+        }
+
+        return new SimpleDateFormat("dd MMM yyyy").format(hearingStartDate);
+    }
+
+    /**
+     * Resets the case data fields relating to creating a referral so that they won't be auto populated when
+     * creating
+     * a new referral.
+     * @param caseData contains all the case data
+     */
+    public static void clearReferralDataFromCaseData(CaseData caseData) {
+        caseData.setReferralHearingDetails(null);
+        caseData.setReferCaseTo(null);
+        caseData.setReferentEmail(null);
+        caseData.setIsUrgent(null);
+        caseData.setReferralSubject(null);
+        caseData.setReferralSubjectSpecify(null);
+        caseData.setReferralDetails(null);
+        caseData.setReferralDocument(null);
+        caseData.setReferralInstruction(null);
+        caseData.setReferredBy(null);
+        caseData.setReferralDate(null);
+    }
+
+    /**
+     * Create fields for referral dropdown selector.
+     * @param caseData contains all the case data
+     */
+    public static DynamicFixedListType populateSelectReferralDropdown(CaseData caseData) {
+        if (CollectionUtils.isEmpty(caseData.getReferralCollection())) {
+            return null;
+        }
+
+        return DynamicFixedListType.from(caseData.getReferralCollection().stream()
+            .filter(r -> !r.getValue().getReferralStatus().equals(ReferralStatus.CLOSED))
+            .map(r -> DynamicValueType.create(
+                r.getValue().getReferralNumber(),
+                r.getValue().getReferralNumber() + " " + r.getValue().getReferralSubject()))
+            .collect(Collectors.toList()));
+    }
+
+    /**
+     * Creates a referral reply and adds it to the referral reply collection.
+     * @param caseData contains all the case data
+     * @param userFullName The full name of the logged-in user
+     */
+    public static void createReferralReply(CaseData caseData, String userFullName) {
+        ReferralType referral = getSelectedReferral(caseData);
+        if (CollectionUtils.isEmpty(referral.getReferralReplyCollection())) {
+            referral.setReferralReplyCollection(new ArrayList<>());
+        }
+        ReferralReplyType referralReply = new ReferralReplyType();
+
+        referralReply.setReplyBy(userFullName);
+        referralReply.setReplyDate(Helper.getCurrentDate());
+        referralReply.setDirectionTo(caseData.getDirectionTo() != null 
+            ? caseData.getDirectionTo() : caseData.getReplyTo());        

```

Comment:
These ternary statements are bit foggy to read. Either add a blank line under them or use some `if` statements